Java 8 Hotspot VM with JDK required to run corfu server?

I fetched a latest build of nsx-corfu-server debian package and installed in a VM. When I invoked the command `corfu-server -ms 9000` to start service, I got the following message:

```
Welcome to CORFU SERVER
Version source (02c02aa)
Serving on port 9000
Service directory: MEMORY mode
Exception in thread "main" java.lang.ExceptionInInitializerError
	at org.corfudb.infrastructure.SequencerServer.<init>(SequencerServer.java:122)
	at org.corfudb.infrastructure.CorfuServer.addSequencer(CorfuServer.java:363)
	at org.corfudb.infrastructure.CorfuServer.main(CorfuServer.java:215)
Caused by: java.lang.IllegalStateException
	at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:268)
	at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:254)
	at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:222)
	at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:208)
	at org.corfudb.util.Utils.<clinit>(Utils.java:48)
	... 3 more
```

A little research brought me to byte-buddy's page [here](https://github.com/raphw/byte-buddy/blob/master/byte-buddy-agent/src/main/java/net/bytebuddy/agent/ByteBuddyAgent.java). Starts from line 268, it says:

>      * <li><b>OpenJDK / Oracle JDK / IBM J9 versions 8-</b>: The installation for HotSpot is only
>      * possible when bundled with a JDK up until Java version 8. It is not possible for runtime-only
>      * installations of HotSpot or J9 for these versions.</li>
>      * </ul>
>      * <p>
>      * If an agent cannot be installed, an {@link IllegalStateException} is thrown.
>      * </p>

I checked my VM and it only had Hotspot VM 1.8.0_102-b14, and there was no JDK installed. It seems like ByteBuddy will not work in this case.

My question is: is there a roundabout to bypass this problem with only JRE available? If not, I will need to add more dependency to my project.

Thanks!

--- a/runtime/pom.xml
+++ b/runtime/pom.xml
@@ -35,11 +35,6 @@
             <version>1.4.24</version>
         </dependency>
         <dependency>
-            <groupId>net.bytebuddy</groupId>
-            <artifactId>byte-buddy-agent</artifactId>
-            <version>1.4.24</version>
-        </dependency>
-        <dependency>
             <groupId>com.esotericsoftware</groupId>
             <artifactId>kryo</artifactId>
             <version>3.0.3</version>

--- a/runtime/src/main/java/org/corfudb/util/Utils.java
+++ b/runtime/src/main/java/org/corfudb/util/Utils.java
@@ -35,30 +35,14 @@ import jdk.internal.org.objectweb.asm.util.TraceMethodVisitor;
 
 import lombok.extern.slf4j.Slf4j;
 
-import net.bytebuddy.agent.ByteBuddyAgent;
-import net.bytebuddy.description.type.TypeDescription;
-import net.bytebuddy.dynamic.ClassFileLocator;
-
 
 /**
  * Created by crossbach on 5/22/15.
  */
 @Slf4j
 public class Utils {
-    private static final Instrumentation instrumentation = ByteBuddyAgent.install();
     private static Printer printer = new Textifier();
     private static TraceMethodVisitor mp = new TraceMethodVisitor(printer);
-
-    public static byte[] getByteCodeOf(Class<?> c) {
-        try {
-            ClassFileLocator locator = ClassFileLocator.AgentBased.of(instrumentation, c);
-            TypeDescription.ForLoadedType desc = new TypeDescription.ForLoadedType(c);
-            ClassFileLocator.Resolution resolution = locator.locate(desc.getName());
-            return resolution.resolve();
-        } catch (IOException ie) {
-            throw new RuntimeException("Couldn't get byte code " + ie.toString(), ie);
-        }
-    }
 
     public static String printByteCode(byte[] bytes) {
         ClassReader cr = new ClassReader(bytes);

